pkg/healthcheck/healthcheck.go

```diff
@@ -1370,6 +1383,30 @@ func (hc *HealthChecker) checkCanGet(namespace, group, version, resource string)
 	return hc.checkCanPerformAction("get", namespace, group, version, resource)
 }
 
+func (hc *HealthChecker) checkAPIService(serviceName string) error {
+	apiServiceClient, err := apiregistrationv1client.NewForConfig(hc.kubeAPI.Config)
+	if err != nil {
+		return err
+	}
+
+	apiStatus, err := apiServiceClient.APIServices().Get(serviceName, metav1.GetOptions{})
+	if err != nil && len(apiStatus.Status.Conditions) == 0 {
+		return err
+	}
+
+	available := false
+	for _, condition := range apiStatus.Status.Conditions {
+		if condition.Status == "True" {
+			available = true
+			break
+		}
+	}
+	if !available {
+		return fmt.Errorf("%s service is not available", serviceName)
```

Comment:
Hey, The APIStatusCondition also has `Reason` and `Message` as `<Reason>: <Message>`, Can you also return them when its not available?

Reply:
I've made the changes